src/Parsers/IParser.cpp

```diff
@@ -45,6 +45,10 @@ void Expected::highlight(HighlightedRange range)
         return;
 
     auto it = highlights.lower_bound(range);
+
+    if (it != highlights.begin())
+        it = std::prev(it);
+
     while (it != highlights.end() && range.begin < it->end)
```

Comment:
Please add a comment. It should be enough to check only one range to the left because it contains non-intersecting ranges ordered by `begin`. So if it's not intersecting with the current one, then all ranges the the left should be fine too